src/bosh-director/bin/bosh-director

```diff
@@ -18,21 +19,23 @@ config = Bosh::Director::Config.load_file(config_file)
 
 route_configuration = Bosh::Director::Api::RouteConfiguration.new(config)
 
-thin_server = Thin::Server.new('127.0.0.1', config.port, signals: false) do
+rack_app = Puma::Rack::Builder.app do
   use Rack::CommonLogger
 
   route_configuration.controllers.each do |route, controller|
     map(route) { run controller }
   end
 end
 
-%w(TERM INT QUIT).each do |signal|
-  trap(signal) do
-    Bosh::Director::Config.logger.info('Shutting down Director')
-    thin_server.stop!
-  end
+puma_configuration = Puma::Configuration.new do |user_config|
+  user_config.workers 3
```

Comment:
Should this be configurable?

Reply:
at some point in the future, probably. Right now, we don't have a clear guidance on when to increase the number of workers and what that would mean for other things, such as your database connection pool. Therefore, I'd like to not expose this number for now.